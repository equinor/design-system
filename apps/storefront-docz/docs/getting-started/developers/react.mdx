---
title: Developers
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
