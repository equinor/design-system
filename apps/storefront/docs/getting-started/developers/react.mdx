---
title: Developers
metaTitle: Developers react
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
