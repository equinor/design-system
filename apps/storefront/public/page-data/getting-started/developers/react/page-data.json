{"componentChunkName":"component---docs-getting-started-developers-react-mdx","path":"/getting-started/developers/react/","result":{"pageContext":{"frontmatter":{"title":"Developers","metaTitle":"Developers react","tabs":["Introduction","CSS framework","React","Figma"],"mode":"draft","route":"/getting-started/developers/react/"},"entry":{"id":"6f9784da00517d57e4627269567073e6","filepath":"docs/getting-started/developers/react.mdx","fullpath":"/Users/vnys/Kode/storefront-replace/apps/storefront/docs/getting-started/developers/react.mdx","route":"/getting-started/developers/react/","slug":"docs-getting-started-developers-react","name":"React","menu":"","headings":[]}}}}