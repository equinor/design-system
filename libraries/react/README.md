@eds-ui/react
===============

1. `npm i -g pnpm`
1. `pnpm install`
1. `pnpm recursive install`

## Run storybook

`pnpm start`

## Build modules

`pnpm recursive run build`
