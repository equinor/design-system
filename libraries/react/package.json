{
  "name": "@eds-ui/react",
  "private": true,
  "version": "0.0.1",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "jest",
    "storybook": "start-storybook"
  },
  "keywords": [],
  "author": "",
  "license": "AGPL",
  "dependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.4",
    "@babel/preset-env": "^7.4.4",
    "@babel/preset-react": "^7.0.0",
    "@storybook/react": "^5.0.11",
    "babel-plugin-styled-components": "^1.10.0",
    "enzyme": "^3.9.0",
    "enzyme-adapter-react-16": "^1.12.1",
    "jest": "^24.8.0",
    "jest-styled-components": "^6.3.1",
    "rollup": "^1.11.3",
    "rollup-plugin-babel": "^4.3.2",
    "rollup-plugin-commonjs": "^9.3.4",
    "rollup-plugin-node-resolve": "^4.2.3",
    "styled-components": "^4.2.0"
  },
  "peerDependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "styled-components": "^4.2.0",
    "prop-types": "^15.7.2"
  },
  "jest": {
    "verbose": true,
    "setupFilesAfterEnv": [
      "<rootDir>/testSetup.js"
    ],
    "testPathIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "transformIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "haste": {
      "providesModuleNodeModules": [
        ".*"
      ]
    },
    "roots": [
      "packages/node_modules"
    ]
  }
}
