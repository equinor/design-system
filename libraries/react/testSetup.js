import React from 'react'
import Enzyme from 'enzyme'
import 'jest-styled-components'
import styled from 'styled-components'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })
