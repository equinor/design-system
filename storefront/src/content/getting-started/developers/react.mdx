---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
mode: draft
---


:warning: Information coming soon
