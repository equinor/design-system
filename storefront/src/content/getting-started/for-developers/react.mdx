---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
---

Developers React
