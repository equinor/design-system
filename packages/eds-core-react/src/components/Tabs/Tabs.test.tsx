/* eslint-disable no-undef */
import { useRef, useEffect, useState, Fragment } from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'
import { axe } from 'jest-axe'
import { Tabs } from '.'

const noop = () => null

const TabsWithRefs = () => {
  const activeRef = useRef<HTMLButtonElement>(null)
  const inactiveRef = useRef<HTMLButtonElement>(null)

  useEffect(() => {
    activeRef.current.textContent = 'Active tab'
    inactiveRef.current.textContent = 'Inactive tab'
  }, [])

  return (
    <Tabs activeTab={0} onChange={noop}>
      <Tabs.List>
        <Tabs.Tab ref={activeRef}>Tab one</Tabs.Tab>
        <Tabs.Tab ref={inactiveRef}>Tab two</Tabs.Tab>
        <Tabs.Tab>Tab three</Tabs.Tab>
      </Tabs.List>
    </Tabs>
  )
}

const TabsWithPanels = ({
  selectedTabIndex = 0,
}: {
  // eslint-disable-next-line react/require-default-props
  selectedTabIndex?: number
}) => {
  const [activeTab, setActiveTab] = useState(selectedTabIndex)

  const handleChange = (index: number) => {
    setActiveTab(index)
  }

  return (
    <Fragment>
      <Tabs activeTab={activeTab} onChange={handleChange}>
        <Tabs.List>
          <Tabs.Tab>Tab one</Tabs.Tab>
          <Tabs.Tab>Tab two</Tabs.Tab>
          <Tabs.Tab disabled>Tab three</Tabs.Tab>
        </Tabs.List>
        <Tabs.Panels>
          <Tabs.Panel>Panel one</Tabs.Panel>
          <Tabs.Panel>Panel two</Tabs.Panel>
          <Tabs.Panel>Panel three</Tabs.Panel>
        </Tabs.Panels>
      </Tabs>
    </Fragment>
  )
}

describe('Tabs', () => {
  it('Matches snapshot', () => {
    const { asFragment } = render(
      <Tabs id="test" onChange={noop}>
        <Tabs.List>
          <Tabs.Tab>Tab one</Tabs.Tab>
          <Tabs.Tab>Tab two</Tabs.Tab>
          <Tabs.Tab>Tab three</Tabs.Tab>
        </Tabs.List>
      </Tabs>,
    )

    expect(asFragment()).toMatchSnapshot()
  })
  it('Should pass a11y test', async () => {
    const { container } = render(
      <Tabs>
        <Tabs.List>
          <Tabs.Tab>Tab one</Tabs.Tab>
          <Tabs.Tab>Tab two</Tabs.Tab>
          <Tabs.Tab>Tab three</Tabs.Tab>
        </Tabs.List>
        <Tabs.Panels>
          <Tabs.Panel>Panel one</Tabs.Panel>
          <Tabs.Panel>Panel two</Tabs.Panel>
          <Tabs.Panel>Panel three</Tabs.Panel>
        </Tabs.Panels>
      </Tabs>,
    )
    expect(await axe(container)).toHaveNoViolations()
  })
  it('Renders a tablist with three tabs', () => {
    render(
      <Tabs onChange={noop}>
        <Tabs.List>
          <Tabs.Tab>Tab one</Tabs.Tab>
          <Tabs.Tab>Tab two</Tabs.Tab>
          <Tabs.Tab>Tab three</Tabs.Tab>
        </Tabs.List>
      </Tabs>,
    )
    expect(screen.getByRole('tablist')).toBeInTheDocument()
    expect(screen.queryAllByRole('tab')).toHaveLength(3)
  })
  it('Merges forwarded and local refs', () => {
    render(<TabsWithRefs />)
    expect(screen.getByText('Active tab')).toBeInTheDocument()
    expect(screen.getByText('Inactive tab')).toBeInTheDocument()
  })
  it('Switches tabpanel when tab is clicked', () => {
    render(<TabsWithPanels />)
    const targetTab = screen.queryByText('Tab two')
    fireEvent.click(targetTab)
    expect(targetTab).toHaveAttribute('aria-selected', 'true')
    expect(screen.queryByText('Panel two')).toBeVisible()
  })
  it('Switches tabpanel when arrow key is clicked', () => {
    render(<TabsWithPanels />)
    const targetTab = screen.queryByText('Tab two')
    const tablist = screen.queryByRole('tablist')
    fireEvent.keyDown(tablist, {
      key: 'ArrowRight',
    })
    expect(targetTab).toHaveAttribute('aria-selected', 'true')
  })
  it('Skips disabled tabs when navigating using arrowkeys', () => {
    render(<TabsWithPanels selectedTabIndex={1} />)
    const targetTab = screen.queryByText('Tab one')
    const tablist = screen.queryByRole('tablist')
    fireEvent.keyDown(tablist, {
      key: 'ArrowRight',
    })
    expect(targetTab).toHaveAttribute('aria-selected', 'true')
  })
  it("Doesn't crash if no children is provided", () => {
    const testId = 'tabs'
    render(<Tabs data-testid={testId} />)
    expect(screen.getByTestId(testId)).toBeDefined()
  })
  it("Doesn't crash if no children is provided to Tabs.Panel", () => {
    const testId = 'tabspanel'
    render(
      <Tabs>
        <Tabs.List>
          <Tabs.Tab>Tab one</Tabs.Tab>
        </Tabs.List>
        <Tabs.Panels>
          <Tabs.Panel data-testid={testId}></Tabs.Panel>
        </Tabs.Panels>
      </Tabs>,
    )
    expect(screen.getByTestId(testId)).toBeDefined()
  })
  it("Doesn't crash if no children is provided to Tabs.Tab", () => {
    const testId = 'tab'
    render(
      <Tabs>
        <Tabs.List>
          <Tabs.Tab data-testid={testId}>Tab one</Tabs.Tab>
        </Tabs.List>
        <Tabs.Panels>
          <Tabs.Panel>Panel one</Tabs.Panel>
        </Tabs.Panels>
      </Tabs>,
    )
    expect(screen.getByTestId(testId)).toBeDefined()
  })
  it("Doesn't crash if no children is provided to Tabs.List or Tabs.Panels", () => {
    const tablist = 'tablist'
    const tabpanels = 'tabpanels'
    render(
      <Tabs>
        <Tabs.List data-testid={tablist}></Tabs.List>
        <Tabs.Panels data-testid={tabpanels}></Tabs.Panels>
      </Tabs>,
    )
    expect(screen.getByTestId(tablist)).toBeDefined()
    expect(screen.getByTestId(tabpanels)).toBeDefined()
  })
})
