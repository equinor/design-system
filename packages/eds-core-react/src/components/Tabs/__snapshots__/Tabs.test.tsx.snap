// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-auto-flow: column;
  grid-auto-columns: max-content;
  overflow-x: hidden;
  -webkit-scroll-snap-type: x mandatory;
  -moz-scroll-snap-type: x mandatory;
  -ms-scroll-snap-type: x mandatory;
  scroll-snap-type: x mandatory;
  overscroll-behavior-x: contain;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  outline: none;
  height: 48px;
  padding-left: 16px;
  padding-right: 16px;
  margin: 0;
  color: rgba(61,61,61,1);
  font-family: Equinor;
  font-size: 1.000rem;
  font-weight: 500;
  line-height: 1.000em;
  -webkit-letter-spacing: 0.013em;
  -moz-letter-spacing: 0.013em;
  -ms-letter-spacing: 0.013em;
  letter-spacing: 0.013em;
  text-align: left;
  color: var(--eds_interactive_primary__resting,rgba(0,112,121,1));
  background-color: transparent;
  position: relative;
  white-space: nowrap;
  text-overflow: ellipsis;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow-x: hidden;
  -webkit-scroll-snap-align: end;
  -moz-scroll-snap-align: end;
  -ms-scroll-snap-align: end;
  scroll-snap-align: end;
  -webkit-scroll-snap-stop: always;
  -moz-scroll-snap-stop: always;
  -ms-scroll-snap-stop: always;
  scroll-snap-stop: always;
  border-bottom: 2px solid var(--eds_ui_background__medium,rgba(220,220,220,1));
  border-bottom: 2px solid var(--eds_interactive_primary__resting,rgba(0,112,121,1));
}

.c1:focus {
  outline: none;
}

.c1[data-focus],
.c1[data-focus-visible-added]:focus {
  outline: 2px dashed var(--eds_interactive__focus,rgba(0,112,121,1));
  outline-offset: -2px;
}

.c1:focus-visible {
  outline: 2px dashed var(--eds_interactive__focus,rgba(0,112,121,1));
  outline-offset: -2px;
}

.c1::-moz-focus-inner {
  border: 0;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  outline: none;
  height: 48px;
  padding-left: 16px;
  padding-right: 16px;
  margin: 0;
  color: rgba(61,61,61,1);
  font-family: Equinor;
  font-size: 1.000rem;
  font-weight: 500;
  line-height: 1.000em;
  -webkit-letter-spacing: 0.013em;
  -moz-letter-spacing: 0.013em;
  -ms-letter-spacing: 0.013em;
  letter-spacing: 0.013em;
  text-align: left;
  color: rgba(61,61,61,1);
  background-color: transparent;
  position: relative;
  white-space: nowrap;
  text-overflow: ellipsis;
  -webkit-text-decoration: none;
  text-decoration: none;
  overflow-x: hidden;
  -webkit-scroll-snap-align: end;
  -moz-scroll-snap-align: end;
  -ms-scroll-snap-align: end;
  scroll-snap-align: end;
  -webkit-scroll-snap-stop: always;
  -moz-scroll-snap-stop: always;
  -ms-scroll-snap-stop: always;
  scroll-snap-stop: always;
  border-bottom: 2px solid var(--eds_ui_background__medium,rgba(220,220,220,1));
}

.c2:focus {
  outline: none;
}

.c2[data-focus],
.c2[data-focus-visible-added]:focus {
  outline: 2px dashed var(--eds_interactive__focus,rgba(0,112,121,1));
  outline-offset: -2px;
}

.c2:focus-visible {
  outline: 2px dashed var(--eds_interactive__focus,rgba(0,112,121,1));
  outline-offset: -2px;
}

.c2::-moz-focus-inner {
  border: 0;
}

@media (prefers-reduced-motion:no-preference) {
  .c0 {
    -webkit-scroll-behavior: smooth;
    -moz-scroll-behavior: smooth;
    -ms-scroll-behavior: smooth;
    scroll-behavior: smooth;
  }
}

@media (hover:none) {
  .c0 {
    overflow-x: scroll;
    -webkit-overflow-scrolling: touch;
    -webkit-scrollbar-width: 0;
    -moz-scrollbar-width: 0;
    -ms-scrollbar-width: 0;
    scrollbar-width: 0;
  }

  .c0 ::-webkit-scrollbar {
    width: 0;
    height: 0;
  }
}

@media (hover:hover) and (pointer:fine) {
  .c1[data-hover],
  .c1:hover {
    color: var(--eds_interactive_primary__hover,rgba(0,79,85,1));
    background: var(--eds_interactive_primary__hover_alt,rgba(222,237,238,1));
    cursor: pointer;
  }
}

@media (hover:hover) and (pointer:fine) {
  .c2[data-hover],
  .c2:hover {
    color: rgba(61,61,61,1);
    background: var(--eds_interactive_primary__hover_alt,rgba(222,237,238,1));
    cursor: pointer;
  }
}

<div>
    <div
      class="c0"
      role="tablist"
    >
      <button
        aria-controls="test-panel-1"
        aria-disabled="false"
        aria-selected="true"
        class="c1"
        id="test-tab-1"
        role="tab"
        tabindex="0"
        type="button"
      >
        Tab one
      </button>
      <button
        aria-controls="test-panel-2"
        aria-disabled="false"
        aria-selected="false"
        class="c2"
        id="test-tab-2"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tab two
      </button>
      <button
        aria-controls="test-panel-3"
        aria-disabled="false"
        aria-selected="false"
        class="c2"
        id="test-tab-3"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Tab three
      </button>
    </div>
  </div>
</DocumentFragment>
`;
